cms/djangoapps/contentstore/utils.py

```diff
@@ -81,20 +83,26 @@ def delete_course_and_groups(course_key, user_id):
 
 def get_lms_link_for_item(location, preview=False):
     """
-    Returns an LMS link to the course with a jump_to to the provided location.
+    Returns an LMS link to the course with a jump_to to the provided location, also checks if
+    Comprehensive Theming is enabled return the LMS link of enabled site's domain name.
 
     :param location: the location to jump to
     :param preview: True if the preview version of LMS should be returned. Default value is false.
     """
     assert isinstance(location, UsageKey)
 
-    if settings.LMS_BASE is None:
-        return None
+    lms_base = None
+    if theming_helpers.is_comprehensive_theming_enabled():
+        lms_base = SiteConfiguration.get_site_domain_for_org(location.org)
 
-    if preview:
-        lms_base = settings.FEATURES.get('PREVIEW_LMS_BASE')
-    else:
-        lms_base = settings.LMS_BASE
+    if lms_base is None:
+        if settings.LMS_BASE is None:
+            return None
+
+        if preview:
+            lms_base = settings.FEATURES.get('PREVIEW_LMS_BASE')
+        else:
+            lms_base = settings.LMS_BASE
```

Comment:
I think we can simplify this change set by looking for SiteConfiguration overrides for `LMS_BASE` and `PREVIEW_LMS_BASE` instead of trying to use the Site model's domain as the base of the URL. The only change we would need is in this function and would look something like:
```
lms_base = configuration_helpers.get_value_for_org(location.org, 'LMS_BASE', settings.LMS_BASE)
preview_base = configuration_helpers.get_value_for_org(location.org, 'PREVIEW_LMS_BASE', settings.FEATURES.get('PREVIEW_LMS_BASE'))

if lms_base is None:
    return None

if preview:
    return preview_base

return lms_base
```